File: src/services/context.ts
import { invoke } from '@tauri-apps/api/core'
import type { ConversationContext } from '@/types/context.types'

export async function getContext(sessionId: string): Promise<ConversationContext> {
  return await invoke<ConversationContext>('get_context', { sessionId })
}

export async function updateContext(
  sessionId: string,
  context: ConversationContext,
): Promise<void> {
  await invoke('update_context', { sessionId, context })
}

File: src-tauri/src/commands/context.rs
use harness_storage::{context_store, ConversationContext};
use tauri::State;

use crate::state::AppState;

#[tauri::command]
pub async fn get_context(
    session_id: String,
    state: State<'_, AppState>,
) -> Result<ConversationContext, String> {
    context_store::load(&state.db, &session_id)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn update_context(
    session_id: String,
    context: ConversationContext,
    state: State<'_, AppState>,
) -> Result<(), String> {
    context_store::save(&state.db, &session_id, &context)
        .await
        .map_err(|e| e.to_string())
}
